error/s2n_errno.c

```diff
@@ -154,6 +154,7 @@ static const char *no_such_error = "Internal s2n error";
     ERR_ENTRY(S2N_ERR_ARRAY_INDEX_OOB, "Array index out of bounds") \
     ERR_ENTRY(S2N_ERR_FREE_STATIC_BLOB, "Cannot free a static blob") \
     ERR_ENTRY(S2N_ERR_RESIZE_STATIC_BLOB, "Cannot resize a static blob") \
+    ERR_ENTRY(S2N_ERR_LIBCRYPTO_NOT_SUPPORTED, "Cannot use this LibCrypto for the requested operation") \
```

Comment:
We should be extra clear about whether this is s2n being incompatible with a specific LibCrypto, or a specific LibCrypto that doesn't support an operation that other LibCrypto's do support. Also, if it's the latter, the error could be clearer as well `S2N_ERR_NO_AVAILABLE_{BORINGSSL/LIBRESSL/OPENSSL}_API`

Reply:
Good point, this is the latter. Updated the error.